zebrad/tests/acceptance.rs

```diff
@@ -103,47 +135,57 @@ where
         }
     }
 
-    fn with_config(self, mut config: ZebradConfig) -> Result<Self> {
+    fn with_config(self, config: &mut ZebradConfig) -> Result<Self> {
+        self.cache_config_update_helper(config)?
+            .write_config_helper(config)
+    }
+
+    fn with_exact_config(self, config: &ZebradConfig) -> Result<Self> {
```

Comment:
This is the new utility function, all the rest is refactoring and type system enforcement

Reply:
Oh and better documentation